agent_server.py

```
@@ -1281,32 +1373,40 @@ async def list_tasks():
-        # terminate all running processes and clear registry
-        for tid, info in list(Modules.task_registry.items()):
-            p = info.get("_proc")
+        # Cancel any in-flight background analyzer tasks
+        for t in list(Modules._background_tasks):
+            if not t.done():
+                t.cancel()
+                try:
+                    await t
+                except asyncio.CancelledError:
+                    pass
+                except Exception as e:
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)